File: insertion sort/insertion-sort.cpp
#include <iostream>
#include <vector>
using namespace std;

void insertionSort(vector<double>& prices) {
    for (int i = 1; i < prices.size(); i++) {
        double key = prices[i];
        int j = i - 1;
        while (j >= 0 && prices[j] > key) {
            prices[j + 1] = prices[j];
            j--;
        }
        prices[j + 1] = key;
    }
}

int main() {
    vector<double> prices = {99.99, 49.95, 299.49, 19.95};
    insertionSort(prices);

    cout << "Sorted prices: ";
    for (double price : prices) {
        cout << price << " ";
    }
    cout << endl;

    return 0;
}

File: insertion sort/insertion-sort.py
def insertion_sort(prices):
    for i in range(1, len(prices)): # 1,2,3
        key = prices[i]
        j = i - 1
        while j >= 0 and prices[j] > key: 
            prices[j + 1] = prices[j]
            j -= 1
        prices[j + 1] = key

# sort prices
prices = [99.99, 49.95, 299.49, 19.95]
insertion_sort(prices)
print("Sorted prices:", prices)
